keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -373,33 +380,42 @@ class ApplicationService(
     deliveryConfig: DeliveryConfig,
     environmentName: String,
     artifact: DeliveryArtifact,
-    version: String,
+    artifactVersion: PublishedArtifact,
     status: PromotionStatus,
-    verifications: List<VerificationSummary>) =
-    buildArtifactSummaryInEnvironment(deliveryConfig, environmentName, artifact, version, status)
+    verifications: List<VerificationSummary>,
+    versions: List<PublishedArtifact>,
+    artifactInfoInEnvironment: Map<String, List<StatusInfoForArtifactInEnvironment>>
+  ) =
+    buildArtifactSummaryInEnvironment(deliveryConfig, environmentName, artifact, artifactVersion, status, versions, artifactInfoInEnvironment)
       ?.copy(verifications = verifications)
 
-  private fun buildArtifactSummaryInEnvironment(deliveryConfig: DeliveryConfig, environmentName: String, artifact: DeliveryArtifact, version: String, status: PromotionStatus): ArtifactSummaryInEnvironment? {
-    val currentArtifact = getArtifactInstance(artifact, version)
-
+  private fun buildArtifactSummaryInEnvironment(
+    deliveryConfig: DeliveryConfig,
+    environmentName: String,
+    artifact: DeliveryArtifact,
+    currentArtifact: PublishedArtifact,
```

Comment:
Consider standardizing on one name for this object. You have `currentArtifact` here and `artifactVersion` on line 383.